lib.js

```diff
@@ -0,0 +1,12 @@
+'use strict'
+
+const _ = require('lodash')
+
+const getErr = err => _.isString(err) ? new Error(err) : err
```

Comment:
both utils are similar so I would opt in to have just getErr and adjust  `if (isPromise) {...` section to use getError when error is not nil